cidata.py

```diff
@@ -561,74 +568,86 @@ def read(
         enabled = podnet_b
         disabled = podnet_a
     elif podnet_a_enabled is True and podnet_b_enabled is True:
-        return False, data_dict, messages[3216]
+        retval = False
+        message_list.append(messages[3216])
     elif podnet_a_enabled is False and podnet_b_enabled is False:
-        return False, data_dict, messages[3217]
+        retval = False
+        message_list.append(messages[3217])
     else:
-        return False, data_dict, messages[3218]
+        retval = False
+        message_list.append(messages[3218])
 
     # define payloads
     read_userdata_payload = f'cat {domain_path}/userdata'
     read_metadata_payload = f'cat {domain_path}/metadata'
 
     # call rcc comms_ssh for metadata retrieval from enabled PodNet
     try:
-        read_metadata, stdout, stderr = comms_ssh(
+        exit_code, stdout, stderr = comms_ssh(
             host_ip=enabled,
             payload=read_metadata_payload,
             username='robot',
         )
     except CouldNotConnectException:
-        return False, data_dict, messages[3221]
+        retval = False
+        message_list.append(messages[3221])
 
-    if read_metadata.exit_code != SUCCESS_CODE:
-        return False, data_dict, messages[3222] + f'{read_metadata.exit_code}s.'
+    if exit_code != SUCCESS_CODE:
```

Comment:
`exit_code` may be undefined if the rcc.comms_ssh() could not connect.

Reply:
Fixed.